src/components/cards/RelationshipsCard.jsx

```diff
@@ -1,106 +1,510 @@
+    });
+    if (response?.status === 204)
+      setConfirmDeleteDialog({
+        realtionshipGuid: null,
+        open: false,
+      });
+  };
+
+  const relationshipCols = [
+    {
+      reference: 'nonSelfFirstName',
+      name: 'nonSelfFirstName',
+      label: 'Individual',
+    },
+    { reference: 'type', name: 'type', label: 'Type' },
+    { reference: 'role', name: 'role', label: 'Role' },
+    {
+      reference: 'actions',
+      name: 'guid',
+      label: 'Actions',
+      options: {
+        customBodyRender: (_, relationship) => {
+          return [
+            <ActionIcon
+              variant="view"
+              href={`/individuals/${relationship?.nonSelfGuid}`}
+              linkProps={{ newTab: true }}
+            />,
+            <ActionIcon
+              variant={'delete'}
+              onClick={() => {
+                setConfirmDeleteDialog({
+                  relationshipGuid: relationship?.guid,
+                  open: true,
+                });
+              }}
+            />,
+          ];
+        },
+      },
+    },
+  ];
+
+  const onChangeType = newType => {
+    setCurrentType(newType);
+    setCurrentRole1(null);
+    setCurrentRole2(null);
+    setCurrentRoles(get(newType, 'roles', []));
+  };
+
+  const onChangeRole1 = newRole => {
+    setCurrentRole1(newRole);
+  };
+
+  const onChangeRole2 = newRole => {
+    setCurrentRole2(newRole);
+  };
+
+  const onCloseDialog = () => {
+    setSelectedIndividualId(null);
+    setSelectedIndividualFirstName(null);
+    setOpenRelationshipDialog(false);
+    setCurrentType(null);
+    setCurrentRole1(null);
+    setCurrentRole2(null);
+    clearPostRelationshipError();
+  };
+
+  const onSubmit = async () => {
+    const response = await postRelationship({
+      individual_1_guid: individualGuid,
+      individual_2_guid: selectedIndividualId,
+      individual_1_role_guid: currentRole1?.guid,
+      individual_2_role_guid: currentRole2?.guid,
+      type_guid: currentType?.guid,
+    });
+    if (response?.status === 200) {
+      onCloseDialog();
+    }
+  };
+
+  return [
+    <ConfirmDelete
+      open={confirmDeleteDialog?.open}
+      onClose={() =>
+        setConfirmDeleteDialog({
+          realtionshipGuid: null,
+          open: false,
+        })
+      }
+      onDelete={async () => {
+        onDelete(confirmDeleteDialog?.relationshipGuid);
+      }}
+      deleteInProgress={deleteRelationshipLoading}
+      error={deleteRelationshipError}
+      onClearError={clearDeleteRelationshipError}
+      messageId="CONFIRM_REMOVE_RELATIONSHIP"
+    />,
+    <StandardDialog
+      open={openRelationshipDialog}
+      onClose={onCloseDialog}
+      titleId={'ADD_RELATIONSHIP'}
+    >
+      <DialogContent>
+        <IndividualSelector
+          setSelectedIndividualId={setSelectedIndividualId}
+          setSelectedIndividualFirstName={
+            setSelectedIndividualFirstName
+          }
+        />
+        <div
+          id="rel-type"
```

Comment:
relationship-type

Reply:
Never thought I'd live to see the day when someone lengthens one of my names. Done.